src/borg/archive.py

```diff
@@ -795,7 +795,7 @@ def process_file(self, path, st, cache, ignore_inode=False):
             # there should be no information in the cache about special files processed in
             # read-special mode, but we better play safe as this was wrong in the past:
             path_hash = ids = None
-        first_run = not cache.files
+        first_run = not cache.files and cache.do_files
```

Comment:
What"s the point of this message anyhow? If it"s to signal the beginning of the archiving op it should be somewhere in Archiver._process instead.